clang/lib/Driver/Driver.cpp

```diff
@@ -4939,6 +4957,12 @@ class OffloadingActionBuilder final {
             auto &TargetInfo = std::get<1>(TargetActionInfo);
             A = C.getDriver().ConstructPhaseAction(C, Args, CurPhase, A,
                                                    AssociatedOffloadKind);
+            // No need to generate the integration header if
+            // -fsycl-no-use-integration-header is specified.
+            if (!Args.hasFlag(options::OPT_fsycl_use_integration_headers,
+                              options::OPT_fno_sycl_use_integration_headers,
+                              true))
```

Comment:
The default value of `true` is passed in multiple places (here, in `Driver::BuildActions()`, in `Clang::ConstructHostCompilerJob()`, in `Clang::ConstructJob()`, and in the suggested change I offered in a previous comment). Perhaps we should define the default in one place somewhere. There are a few different approaches. I don't have a strong opinion on any of these, but I think we should do something.
- We could introduce a constant that is passed in each of the calls to `hasFlag()`.
- We could introduce a function to be called at each of these locations to ascertain whether integration headers are being used; that function would query the options and supply the default.
- We could check the options in one place and store that result.

Reply:
I went with option 2.   Thanks @tahonermann

Reply:
Great! It looks like this particular location didn't get updated to use it though.